src/transport/mqtt/config/tls.rs

```diff
@@ -107,10 +108,17 @@ impl ClientAuth {
         Ok(Some(ClientAuth { private_key, certs }))
     }
 
-    pub(crate) async fn tls_config(self) -> Result<rustls::ClientConfig, PairingError> {
-        let roots = read_root_cert_store().await?;
-
-        rustls::ClientConfig::builder()
+    pub(crate) async fn tls_config(
+        self,
+        roots: Arc<RootCertStore>,
+    ) -> Result<rustls::ClientConfig, PairingError> {
+        let crypto_provider = CryptoProvider::get_default()
+            .cloned()
+            .unwrap_or_else(|| Arc::new(rustls::crypto::aws_lc_rs::default_provider()));
+
+        rustls::ClientConfig::builder_with_provider(crypto_provider)
+            .with_safe_default_protocol_versions()
```

Comment:
Separate this into a function (not a method of client auth) to share with the TransportProvider:

```rust
pub(crate) fn tls_client_config(roots: Arc<RootCertStore>) -> Result<rustls::ConfigBuilder<ClientConfig, WantsCerts>, PairingError> {
     let crypto_provider = CryptoProvider::get_default()
            .cloned()
            .unwrap_or_else(|| Arc::new(rustls::crypto::aws_lc_rs::default_provider()));
    
     
        rustls::ClientConfig::builder_with_provider(crypto_provider)
            .with_safe_default_protocol_versions()
            .map_err(PairingError::Tls)?
            .with_root_certificates(roots)
}
```

Reply:
Also  the insecure one need to be added the same way. Additionally passing the CrytpoProvider to that too